### PR TITLE
Checkstyle

### PR DESCRIPTION
**Pull request by YTG1234**
~~This branch was created from the `mc1.16/fabric/kotlin_dsl` branch, so that one must be merged before we can merge this one.~~

**Progress**
- [X] Create Checkstyle configuration file
- [X] Make sure the config works
- [X] Add our own checks
~~- [ ] Reformat code according to the config~~ - CANCELLED

### CODE REVIEW

**Comment by YTG1234** [comment]
Note: Because this was created from the Kotlin DSL branch, it includes its diff too. Just ignore it for now.

**Comment by YTG1234** [comment]
The only parts you should actually review are:
https://github.com/Create-Fabric/Create-Refabricated/blob/414396b89b560b81ffaafc89f37e3dfb7a2cf244/build.gradle.kts#L18-L109

https://github.com/Create-Fabric/Create-Refabricated/blob/414396b89b560b81ffaafc89f37e3dfb7a2cf244/settings.gradle.kts#L10-L11

https://github.com/Create-Fabric/Create-Refabricated/tree/mc1.16/fabric/checkstyle/checkstyleChecks

https://github.com/Create-Fabric/Create-Refabricated/blob/414396b89b560b81ffaafc89f37e3dfb7a2cf244/checkstyle/checkstyle.xml#L1-L83

File: .editorconfig [hunk 1]
```diff
@@ -106,7 +106,7 @@ ij_java_for_statement_wrap = off
 ij_java_generate_final_locals = false
 ij_java_generate_final_parameters = false
 ij_java_if_brace_force = never
-ij_java_imports_layout = *,|,javax.**,java.**,|,$*
+ij_java_imports_layout = java.**,|,javax.**,|,*,|,net.minecraft.**,|,org.spongepowered.**,|,net.fabricmc.**,|,$*
```
Comment (PepperCode1) [inline .editorconfig]:
```suggestion
ij_java_imports_layout = $*,|,java.**,|,javax.**,|,*,|,net.minecraft.**,|,org.spongepowered.**,|,net.fabricmc.**,|,me.pepperbell.**,|,com.simibubi.**
```

File: checkstyle/checkstyle.xml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+	<!-- Imports -->
+	<module name="TreeWalker">
+		<module name="AvoidStarImport">
+			<property name="allowStaticMemberImports" value="true" />
+		</module>
+
+		<module name="ImportOrder">
+			<property name="option" value="top" />
+			<property name="groups" value="java,javax,*,net.minecraft,org.spongepowered,net.fabricmc,com.pepperbell,com.simibubi" />
```
Comment (PepperCode1) [inline checkstyle/checkstyle.xml]:
```suggestion
			<property name="groups" value="java,javax,*,net.minecraft,org.spongepowered,net.fabricmc,me.pepperbell,com.simibubi" />
```

**Review by PepperCode1 — CHANGES_REQUESTED** [review]
Otherwise, LGTM.

**Comment by YTG1234** [comment]
I'll resolve the conflicts and merge this then

**Review by PepperCode1 — APPROVED** [review]

**Pull request merged**